web/src/components/network/WifiNetworksListPage.tsx

```diff
@@ -44,21 +44,28 @@ import {
   Stack,
 } from "@patternfly/react-core";
 import { generatePath } from "react-router-dom";
-import { useQueryClient } from "@tanstack/react-query";
 import { Icon } from "~/components/layout";
-import { WifiConnectionForm } from "~/components/network";
-import { ButtonLink } from "~/components/core";
-import { DeviceState } from "~/client/network/model";
-import { formatIp } from "~/client/network/utils";
-import { useInstallerClient } from "~/context/installer";
-import { useSelectedWifi, useSelectedWifiChange } from "~/queries/network";
+import { ButtonLink, EmptyState } from "~/components/core";
+import WifiConnectionForm from "~/components/network/WifiConnectionForm";
 import { PATHS } from "~/routes/network";
+import { DeviceState, WifiNetwork, Wireless } from "~/types/network";
 import { _ } from "~/i18n";
+import { formatIp } from "~/utils/network";
+import {
+  useNetwork,
+  useRemoveConnectionMutation,
+  useSelectedWifi,
+  useSelectedWifiChange,
+  useWifiNetworks,
+} from "~/queries/network";
+import { slugify } from "~/utils";
+import { connect, disconnect } from "~/api/network";
 
-const HIDDEN_NETWORK = Object.freeze({ hidden: true });
+type HiddenNetwork = { hidden: boolean };
```

Comment:
this type quite smell for me as I expect that hidden network contain more entries...why WifiNetwork cannot be used? or maybe instead of const variable method that do that simple check for hidden value on common WifiNetwork type can be used?

Reply:
Let me revisit this and see how we can improve it.

Reply:
Ok, I see now. WifiNetwork is an AccessPoint plus WifiNetworkStatus plus other optional values like `hidden`. So, HIDDEN_NETWORK cannot be a WifiNetwork sinde it does belong to none access point. A bit lost here, so I prefere to wait until @teclator can have a look.